src/docker/base/Dockerfile

```diff
@@ -1,45 +1,38 @@
 # Copyright 2023 Flower Labs GmbH. All Rights Reserved.
 
-ARG UBUNTU_VERSION=22.04
-FROM ubuntu:$UBUNTU_VERSION as base
+ARG PYTHON_VERSION=3.10
+ARG ALPINE_VERSION=alpine3.19
+FROM python:${PYTHON_VERSION}-${ALPINE_VERSION} as base
 
-ENV DEBIAN_FRONTEND noninteractive
+ENV DEBIAN_FRONTEND=noninteractive \
 # Send stdout and stderr stream directly to the terminal. Ensures that no
 # output is retained in a buffer if the application crashes.
-ENV PYTHONUNBUFFERED 1
+  PYTHONUNBUFFERED=1 \
 # Typically, bytecode is created on the first invocation to speed up following invocation.
 # However, in Docker we only make a single invocation (when we start the container).
 # Therefore, we can disable bytecode writing.
-ENV PYTHONDONTWRITEBYTECODE 1
+  PYTHONDONTWRITEBYTECODE=1 \
 # Ensure that python encoding is always UTF-8.
-ENV PYTHONIOENCODING UTF-8
-ENV LANG C.UTF-8
-ENV LC_ALL C.UTF-8
+  PYTHONIOENCODING=UTF-8 \
+  LANG=C.UTF-8 \
+  LC_ALL=C.UTF-8
 
 # Install system dependencies
-RUN apt-get update \
-    && apt-get -y --no-install-recommends install \
-    clang-format git unzip ca-certificates openssh-client liblzma-dev \
-    build-essential libssl-dev zlib1g-dev libbz2-dev libreadline-dev wget\
-    libsqlite3-dev curl llvm libncursesw5-dev xz-utils tk-dev libxml2-dev \
-    libxmlsec1-dev libffi-dev liblzma-dev \
-    && rm -rf /var/lib/apt/lists/*
-
-# Install PyEnv and Python
-ARG PYTHON_VERSION
-ENV PYENV_ROOT /root/.pyenv
-ENV PATH $PYENV_ROOT/shims:$PYENV_ROOT/bin:$PATH
-# https://github.com/hadolint/hadolint/wiki/DL4006
-SHELL ["/bin/bash", "-o", "pipefail", "-c"]
-RUN curl -L https://github.com/pyenv/pyenv-installer/raw/master/bin/pyenv-installer | bash
-RUN pyenv install ${PYTHON_VERSION} \
-    && pyenv global ${PYTHON_VERSION} \
-    && pyenv rehash
+RUN apk add --no-cache \
+    git=2.43.0-r0 \
+    py3-pip=23.3.1-r0
 
 # Install specific version of pip
 ARG PIP_VERSION
-RUN python -m pip install --no-cache-dir pip==$PIP_VERSION
-
-# Install specific version of setuptools
 ARG SETUPTOOLS_VERSION
-RUN python -m pip install --no-cache-dir setuptools==$SETUPTOOLS_VERSION
+RUN pip install --no-cache-dir \
+    pip==${PIP_VERSION:-24.0} \
```

Comment:
Just my personal preference, but I would avoid using default settings here. I'm fine with having defaults for Alpine and Python version as those only belong/are relevant in the Dockerfile, but for PIP_VERSION, SETUPTOOLS_VERSION, etc. we already specify the version [here](https://github.com/adap/flower/blob/main/.github/actions/bootstrap/action.yml) and pass them when building the [image](https://github.com/adap/flower/blob/main/.github/workflows/docker-base.yml).

Reply:
I found it easier to not specify all versions manually for local building and testing, although the versions are defined in the GitHub action.

We can use this as a builder image and then copy over all build modules to the final container.

Reply:
See a similar use case [here](https://github.com/neXenio/kapitan/blob/master/Dockerfile).